entry_types/scrolled/package/spec/frontend/MediaPlayer-spec.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect'
+import 'support/mediaElementStub';
+
+import {renderInEntry} from 'support';
+
+import {MediaPlayer} from 'frontend/MediaPlayer';
+
+describe('MediaPlayer', () => {
+  let audioSources =[ 
+    {type: 'audio/ogg', src: 'http://example.com/example.ogg'},
+    {type: 'audio/m4a', src: 'http://example.com/example.m4a'},
+    {type: 'audio/mp3', src: 'http://example.com/example.ogg'}
+  ];
+
+  it('do not renders audio tag without sources', () => {
+    const {queryByRole} =
+      renderInEntry(<MediaPlayer type={'audio'} />, {
+        seed: {}
+      });
+
+    expect(queryByRole('audio')).toBeNull();
+  });
+
+  it('renders audio tag for audio sources', () => {
+    const {queryByRole} =
+      renderInEntry(<MediaPlayer type={'audio'} sources={audioSources} />, {
+        seed: {}
+      });
+
+    expect(queryByRole('audio')).toBeDefined();
```

Comment:
queryByRole('..') is somehow always returning null. What could be the reason of it?

Reply:
I don't know. Maybe you can inspect the rendered result by looking at `container.innerHTML` (https://testing-library.com/docs/react-testing-library/api#container)?

Reply:
If I dont use renderInEntry then i do get something like this

`<body class="has_autoplay_support has_css_animations has_no_facebook_toolbar has_no_ie has_no_ios_platform has_no_iphone_platform has_no_mobile_platform has_no_phone_platform has_pushstate_support has_request_animation_frame_support has_no_touch_support has_rewrite_video_sources_support has_stop_buffering_support has_buffer_underrun_waiting_support has_prebuffering_support has_no_mp4_support_only has_no_mse_and_native_hls_support has_no_native_video_player has_volume_control_support has_audio_context_volume_fading_support"><div><div aria-label="Audio Player" class="vjs-paused vjs-controls-disabled vjs-audio vjs-workinghover vjs-v6 vjs-user-active video-js player" id="pageflow_media_element_0" lang="en-us" role="region" tabindex="-1"><audio class="vjs-tech" id="pageflow_media_element_0_html5_api" tabindex="-1" /><div aria-disabled="false" class="vjs-poster vjs-hidden" tabindex="-1" /><div aria-atomic="true" aria-live="off" class="vjs-text-track-display" /></div></div></body>`

Reply:
> If I dont use renderInEntry 

What does this mean? `container` should also be usable in combination with `renderInEntry`.

Reply:
I dont know about this API, I can try

Reply:
it is working now